activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -50,7 +50,7 @@ def visit_TableDefinition(o)
             end
 
             if supports_foreign_keys?
-              statements.concat(o.foreign_keys.map { |to_table, options| foreign_key_in_create(o.name, to_table, options) })
+              statements.concat(o.foreign_keys.map { |fk| foreign_key_in_create(o.name, fk[0], fk[1]) })
```

Comment:
You're right; the original syntax works. I didn't realise that you could pass two parameters to the block like that when iterating over a nested array.

Reverted and rebased.